client/dashboard/sites/settings-site-visibility/trial-upsell-notice.tsx

```diff
@@ -0,0 +1,67 @@
+import { Button } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { useAnalytics } from '../../app/analytics';
+import Notice from '../../components/notice';
+import {
+	isSiteOnECommerceTrial as getIsSiteOnECommerceTrial,
+	isSiteOnMigrationTrial as getIsSiteOnMigrationTrial,
+	isSiteLaunchable,
+} from '../../utils/site-plans';
+import type { Site } from '../../data/types';
+
+export default function TrialUpsellNotice( { site }: { site: Site } ) {
+	const { recordTracksEvent } = useAnalytics();
+	const isSiteOnECommerceTrial = getIsSiteOnECommerceTrial( site );
+	const isSiteOnMigrationTrial = getIsSiteOnMigrationTrial( site );
+	const isLaunchable = isSiteLaunchable( site );
+
+	useEffect( () => {
+		if ( ! isLaunchable ) {
+			recordTracksEvent( 'calypso_settings_trial_upsell_notice_impression' );
```

Comment:
Not for this PR, but I wonder if a `<RecordImpression>` component would be useful. That way you don't have to worry about cases like this where `isLaunchable === false` returns `null`. You can just return

```tsx
<>
	<RecordImpression name="calypso_settings_trial_upsell_notice_impression" />
	<Notice { ... } />
</>
```

Because you know if you go this far you're confident the impression has occured.